src/utils/Paginator/paginator.js

```diff
@@ -1,38 +1,52 @@
-import React from 'react';
+import React, { useState } from 'react';
+import classnames from 'classnames';
+import { useTranslation } from 'react-i18next';
+
+import { I18N_KEYS } from '../constants';
 
 import styles from './paginator.module.css';
 
-const paginator = (props) => {
+const Paginator = (props) => {
   const {
-    totalUsersCount, currentPage, onPageChanged, pageSize,
+    totalItemsCount, currentPage, onPageChanged, pageSize, portionSize = 10,
   } = props;
 
-  const pagesCount = Math.ceil(totalUsersCount / pageSize);
+  const { t } = useTranslation();
+  const [portionNumber, setPortionNumber] = useState(1);
+  const pagesCount = Math.ceil(totalItemsCount / pageSize);
```

Comment:
Could the next 4 functions extracted into a utils function?
E.g. `const { leftPortionPageNumber ....} = calculatePageStats(portionNumber, totalItemsCount....);`

Reply:
no, this is a paginator utility in itself, and will be reused entirely